lib/parse.js

```diff
@@ -46,14 +46,64 @@ const DEFAULT_VAR_PARSER = {
   image: {
     var: parseSelect,
     advanced: parseVarXStyle
-  }
+  },
+  number: parseRange,
+  range: parseRange
 };
 
 const DEFAULT_VAR_VALIDATOR = {
-  checkbox: validateCheckbox
+  checkbox: validateCheckbox,
+  number: validateRange,
+  range: validateRange
 };
 
 const MANDATORY_META = ['name', 'namespace', 'version'];
+const RANGE_PROPS = ['default', 'min', 'max', 'step'];
+
+function parseRange(state) {
+  parseJSON(state);
+  const result = {
+    min: null,
+    max: null,
+    step: null,
+    units: null
+  };
+  if (typeof state.value === 'number') {
+    result.default = state.value;
+  } else if (Array.isArray(state.value)) {
+    let i = 0;
+    for (const item of state.value) {
+      if (typeof item === 'string') {
+        if (result.units != null) {
+          throw new ParseError({
+            code: 'invalidRangeMultipleUnits',
+            message: 'units is alredy defined',
+            index: state.valueIndex
+          });
+        }
```

Comment:
We'll need to change the UserCSS-authors wiki since it states:

> Only the first units value entry will be used, any additional (non numeric) entries will be ignored, e.g. `[5, 0, 10, 1, 'px', 'em']` will only set the units to `px`. The `em` entry is removed.